src/indicators/ipc.R

```diff
@@ -0,0 +1,238 @@
+library(Ripc)
+library(tidyverse)
+library(countrycode)
+
+output_dir <- file.path(
+  Sys.getenv("AA_DATA_DIR"),
+  "private",
+  "exploration",
+  "glb",
+  "global_monitoring",
+  "outputs"
+)
+
+#############
+#### IPC ####
+#############
+
+# get into country level form
+df_ipc <- ipc_download() %>%
+  mutate(
+    iso3 = countrycode(country, origin = "country.name", destination = "iso3c"),
+    phase_4pl_num = phase_4_num + phase_5_num,
+    phase_4pl_pct = phase_4_pct + phase_5_pct
+  ) %>%
+  group_by(
+    country,
+    iso3,
+    date_of_analysis,
+    analysis_period_start,
+    analysis_period_end,
+    analysis_type
+  ) %>%
+  summarize(
+    across(
+      .cols = ends_with("_pct"),
+      .fns = weighted.mean,
+      w = population,
+      na.rm = TRUE
+    ),
+    across(
+      .cols = c(ends_with("_num"), population),
+      .fns = sum,
+      na.rm = TRUE
+    ),
+    .groups = "drop"
+  )
+
+# get differences between current values
+df_cur_delta <- df_ipc %>%
+  filter(
+    analysis_type == "current"
+  ) %>%
+  group_by(
+    country,
+    iso3
+  ) %>%
+  arrange(
+    date_of_analysis,
+    .by_group = TRUE
+  ) %>%
+  mutate(
+    across(
+      matches("pct$"),
+      ~ .x - lag(.x)
+    ),
+    flag_ipc_curr_3pl = phase_3pl_pct > 0,
```

Comment:
would be less ambiguous if the other # subtracted (last or projected) were included in the names